File: 2-Ad-Hoc/ex2377.c
#include <stdio.h>

int main(){
    //variaveis
    int compriEstrada, distPedagio;
    int custoPorKm, valorPedagio;
    int custTotal = 0;

    //entrada
    scanf("%d %d", &compriEstrada, &distPedagio);
    scanf("%d %d", &custoPorKm, &valorPedagio);

    //processamento
    custTotal = custoPorKm * compriEstrada;
    custTotal += (compriEstrada / distPedagio) * valorPedagio;

    //saida
    printf("%d\n", custTotal);
    return 0;
}

File: 2-Ad-Hoc/Ex2377.java
import java.util.Scanner;

public class Ex2377 { // para ser aceito no URI mude o nome da classe para Main
    public static void main(String[] args) {
        // variaveis & objetos
        Scanner scanner = new Scanner(System.in);
        int compriEstrada, distPedagio;
        int custoPorKm, valorPedagio;
        int custTotal = 0;

        // entrada
        compriEstrada = scanner.nextInt();
        distPedagio = scanner.nextInt();
        custoPorKm = scanner.nextInt();
        valorPedagio = scanner.nextInt();

        // processamento
        custTotal = custoPorKm * compriEstrada;
        custTotal += (compriEstrada / distPedagio) * valorPedagio;

        // saida
        System.out.println(custTotal);

        scanner.close();
    }
}